_source/_posts/2023-04-19-api-integrations.md

```diff
@@ -0,0 +1,105 @@
+
+**When to use API tokens**
+
+Due to these constraints, API tokens are never the preferred authorization method for Okta's APIs. If you're ever in a situation where you "have to" use API tokens in production because OAuth or a service integration won't work, we'd love to hear about it! Let us know in the comments below, or discuss the challenges with your Okta representative. 
+
+## OAuth 2.0 Client Credentials
+
+OAuth has a standard flow for machine-to-machine communication called Client Credentials, the preferred way for processes without user input to communicate to Okta APIs. Using OAuth mechanisms enables you to use a well-known standardized format with fantastic support in libraries, frameworks, and languages of your choice. You can create an API Services application in the Okta Admin Console or use the [Okta CLI](https://cli.okta.com/).
+
+If you haven't used OAuth Client Credentials before, a quick overview is as follows:
+  1. You'll create the Okta API Services application and configure the Okta API Scopes you want this application to have. When you make the Okta application, you'll receive a Client ID and a Client Secret to add to your Okta configuration in your program.
+  2. Your program calls the authorization server's token endpoint to exchange the Client ID and Client Secret for an access token. The access token contains the Okta API Scopes allowed. There may be [Okta SDKs](https://developer.okta.com/code/) that help support this token call for you, depending on your language and framework.
+  > **Note**
+  > If you have a custom authorization server, such as when using a Okta Developer Edition org, you'll have to use signed JSON Web Token (JWT) using a JSON Web Key Set (JWKS). Read more about how to do this on the [Implement OAuth for Okta](https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/main/) developer documentation.
+  3. When you use the access token to call Okta APIs, you'll add the token to the Authorization header using the `Bearer` scheme.
+
+```http
+GET /api/v1/groups?limit=10 HTTP/1.1
+Authorization: Bearer access_token
+Accept: application/json
+Host: dev-1234567.okta.com
+```
+
+**Considerations with using OAuth**
+
+The preferred method is using OAuth to call Okta APIs. It's more secure, and using widely available standards makes the calling process require less manual code. You'll configure the scopes you want the process to access, which aren't tied to a specific user. But it is a 2-step process to make the API call as first you'll need to get the access token. 
```

Comment:
I'm not sure what this should say: "But it is a 2-step process to make the API call as first you'll need to get the access token."

Reply:
I changed it to 
> But you'll need to get the access token before making the API call.